COM-93 Implement dispensation reports

tmvumbi2 opened the pull request:
Jira ticket: https://jembiprojects.jira.com/browse/COM-93

tmvumbi2 committed:
COM-93 Implement dispensation reports
Implements dispensation reports

lauravignoli commented on reportssql/dispensation_report_functions.sql:
```diff
@@ -0,0 +1,147 @@
+        AND cn.name = p_arvName
+        AND o.scheduled_date BETWEEN p_startDate AND p_endDate
+        AND drugOrderIsDispensed(o.patient_id, o.order_id)
+        AND patientPickedARVDrugDuringReportingPeriodWithNoTherapeuticLine(o.patient_id, p_startDate, p_endDate)
```
@tmvumbi2 Please check use of this function as it doesn't check that there's no therapeutic line.

lauravignoli replied:
Please rename and add an extra condition to check that the patient doesn't have a therapeutic line.

lauravignoli replied:
Checked with Richard and for AES and PA, we dont care about checking for therapeutic line. But we should probably rename the function to avoid confusion.

lauravignoli commented on reportssql/dispensation_report_functions.sql:
```diff
@@ -0,0 +1,147 @@
+        AND o.scheduled_date BETWEEN p_startDate AND p_endDate
+        AND drugOrderIsDispensed(o.patient_id, o.order_id)
+        AND patientPickedARVDrugDuringReportingPeriodWithNoTherapeuticLine(o.patient_id, p_startDate, p_endDate)
+        AND patientReasonForConsultationIsUnplannedAid(o.patient_id)
```
@tmvumbi2 In the spec it says: If it is a PA patient then the reason for consultation must be “Unplanned Aid”. Shouldn't this check be in getNumberOfPA_ARVDispensed instead of here?

lauravignoli replied:
@tmvumbi2 Please apply this condition in getNumberOfPA_ARVDispensed. See spec: If it is a PA patient then the reason for consultation must be “Unplanned Aid”.

Note that Richard updated the spec indicator numbers to:

AES = Accidents of Exposure (PECG Indicator 20)
PA = Punctual Aid (PECG Indicator 19)

lauravignoli committed:
COM-93 Add comments to pecg ind 19 and 20

lauravignoli commented on reportssql/dispensation_report_functions.sql:
```diff
@@ -0,0 +1,147 @@
+END$$ 
+DELIMITER ;
+
+DROP FUNCTION IF EXISTS getNumberOfPA_ARVDispensed;
```
@tmvumbi2 Note that Richard updated the spec indicator numbers to:

AES = Accidents of Exposure (PECG Indicator 20)
PA = Punctual Aid (PECG Indicator 19)

Please update the names accordingly.

lauravignoli commented on reportssql/dispensation_report_functions.sql:
```diff
@@ -0,0 +1,147 @@
+    WHERE o.voided = 0
+        AND drugIsARV(d.concept_id)
+        AND cn.name = p_arvName
+        AND o.scheduled_date BETWEEN p_startDate AND p_endDate
```
@tmvumbi2 Considering that this function is AES and not PA, patientHasNotBeenEnrolledIntoHivProgram should be applied here?

lauravignoli commented on reportssql/dispensation_report_functions.sql:
```diff
@@ -0,0 +1,147 @@
+DROP FUNCTION IF EXISTS getNumberOfAES_ARVDispensed;
+
+DELIMITER $$
+CREATE FUNCTION getNumberOfAES_ARVDispensed(
```
@tmvumbi2 Considering that this function is PA and not AES, patientHasEnrolledIntoHivProgramDuringReportingPeriod should be applied here?

tmvumbi2 committed:
COM-93 Refactor: rename functions
Renames patientPickedARVDrugDuringReportingPeriod to patientWithTherapeuticLinePickedARVDrugDuringReportingPeriod: this function is checking that the patient has picked up an ARV and also that they are in therapeutic line.

Renames patientPickedARVDrugDuringReportingPeriodWithNoTherapeuticLine to patientPickedARVDrugDuringReportingPeriod: this function is only checking that the patient has picked up an ARV (irrespective to whether or not they are in a therapeutic line)

tmvumbi2 committed:
Merge branch 'COM-93' of https://github.com/jembi/cameroon-bahmni-metadata into COM-93

tmvumbi2 committed:
COM-93 Rewrite functions to calculate the number of AES and PA
Updates the logic of the above functions to match the updated requirement
